pyuvsim/simsetup.py

```diff
@@ -745,9 +759,63 @@ def parse_time_params(time_params):
     return_dict['Nspws'] = 1
     return_dict['Npols'] = 4
 
+    time_params = _time_params  # Restore backup
+
     return return_dict
 
 
+def freq_array_to_params(freq_array):
+    """
+    Give the channel width, bandwidth, start, and end frequencies corresponding
+    to a given frequency array.
+
+    Args:
+        freq_array : (ndarray, shape = (Nfreqs,)) of frequencies.
+
+    Returns:
+        Dictionary of frequency parameters consistent with freq_array.
+    """
+    freq_array = np.asarray(freq_array).ravel()
+
+    fdict = {}
+    if freq_array.size < 2:
+        raise ValueError("Frequency array must be longer than 1 to give meaningful results.")
+
+    fdict['channel_width'] = np.diff(freq_array)[0]
+    fdict['Nfreqs'] = freq_array.size
+    fdict['bandwidth'] = fdict['channel_width'] * fdict['Nfreqs']
+    fdict['start_freq'] = freq_array[0]
+    fdict['end_freq'] = freq_array[-1]
+
+    return fdict
+
+
+def time_array_to_params(time_array):
```

Comment:
Where are these functions used? I only see them used in the tests. The reason I ask, is that while the frequencies have to be evenly spaced for uvfits & miriad, time arrays do not in general have to be equally spaced. So if you're using this function to get params from UVData objects, you shouldn't necessarily assume that they will be evenly spaced.

Reply:
There have been a few times I've had an array of frequencies or times and wished I had a quick way to get the right parameters to make it.

I'll adjust the `time_array_to_params` to check if the parameters are evenly spaced first. If it is, it will return some more compact representation of the time array (e.g., start_time + Ntimes + integration_time). If not, it will just return the full array.

Reply:
(To add -- these functions will be used in the `uvdata_to_config` function)

Reply:
that sounds good.